R/simple_ensemble.R

```
@@ -0,0 +1,140 @@
+                            output_type_col = "output_type",
+                            output_type_id_col = "output_type_id",
+                            hub_connection = NULL) {
+  model_out_cols <- colnames(model_outputs)
+  if (!is.data.frame(model_outputs)) {
+    cli::cli_abort(c("x" = "{.arg model_outputs} must be a `data.frame`."))
+  }
+
+  non_task_cols <- c("model_id", output_type_col, output_type_id_col, "value")
+  if (is.null(task_id_cols)) {
+    task_id_cols <- model_out_cols[!model_out_cols %in% non_task_cols]
+  }
+
+  req_col_names <- c(non_task_cols, task_id_cols)
+  if (!all(req_col_names %in% model_out_cols)) {
+    cli::cli_abort(c(
+      "x" = "{.arg model_outputs} did not have all required columns
+             {.val {req_col_names}}."
+    ))
+  }
+
+  ## Validations above this point to be relocated to hubUtils
+  # hubUtils::validate_model_output_df(model_outputs)
+
+  if (nrow(model_outputs) == 0) {
+    cli::cli_warn(c("!" = "{.arg model_outputs} has zero rows."))
+  }
+
+  valid_types <- c("mean", "median", "quantile", "cdf", "pmf")
+  unique_types <- unique(model_outputs[[output_type_col]])
+  invalid_types <- unique_types[!unique_types %in% valid_types]
+  if (length(invalid_types) > 0) {
+    cli::cli_abort(c(
+      "x" = "{.arg model_outputs} contains unsupported output type.",
+      "i" = "Included output type{?s}: {.val {invalid_types}}.",
+      "i" = "Supported output types: {.val {valid_types}}."
+    ))
+  }
+
+  if (is.null(weights)) {
+    agg_args <- c(agg_args, list(x = quote(.data[["value"]])))
+  } else {
+    req_weight_cols <- c("model_id", "weight")
+    if (!all(req_weight_cols %in% colnames(weights))) {
+      cli::cli_abort(c(
+        "x" = "{.arg weights} did not include required columns
+               {.val {req_weight_cols}}."
+      ))
+    }
+
+    weight_by_cols <- colnames(weights)[colnames(weights) != "weight"]
+
+    if ("value" %in% weight_by_cols) {
+      cli::cli_abort(c(
```

Should we abort in this situation or just ignore/remove the `value` column in weights with a warning?

---

I think we should abort -- if `"value"` is in these columns, it's likely that the user either has a conceptual error (trying to weight by value) or a bug.  It seems better to raise an error than to try to proceed and maybe generate unintended results.